tests/topotests/ospf_single_switch/test_ospf_single_switch.py

```diff
@@ -0,0 +1,168 @@
+
+    # Create routers
+    for router_id in range(1, N_ROUTERS + 1):
+        router = tgen.add_router(f"r{router_id}")
+        switch.add_link(router)
+
+        # The shared network needs to be connected to something
+        dummy = tgen.add_switch(f"s{router_id}")
+        dummy.add_link(router)
+
+
+def setup_module(mod):
+    logger.info("OSPF single switch:\n {}".format(TOPOLOGY))
+
+    tgen = Topogen(build_topo, mod.__name__)
+    tgen.start_topology()
+
+    # Starting Routers
+    router_list = tgen.routers()
+
+    for rname, router in router_list.items():
+        logger.info("Loading router %s" % rname)
+        router.load_frr_config(os.path.join(CWD, "{}_frr.conf".format(rname)))
+
+    # Initialize all routers.
+    tgen.start_router()
+
+
+def teardown_module(mod):
+    "Tear-down the test environment"
+    tgen = get_topogen()
+    tgen.stop_topology()
+
+
+def test_all_routes_advertised():
+    tgen = get_topogen()
+
+    if tgen.routers_have_failure():
+        pytest.skip("Skipped because of router(s) failure")
+
+    # networks advertised by each router and the expected next hops
+    networks = {
+        "r1": ("10.0.1.0/24", "120.0.0.1"),
+        "r2": ("10.0.2.0/24", "120.0.0.2"),
+        "r3": ("10.0.3.0/24", "120.0.0.3"),
+        "r4": ("10.0.4.0/24", "120.0.0.4"),
+        "r5": ("10.0.5.0/24", "120.0.0.5"),
+        "r6": ("10.0.6.0/24", "120.0.0.6"),
+        "r7": ("10.0.7.0/24", "120.0.0.7"),
+        "r8": ("10.0.8.0/24", "120.0.0.8"),
+    }
+
+    for router_orig in tgen.routers().keys():
+        for router_dest, network in networks.items():
+            if router_orig != router_dest:
+                check_route(router_orig, network[0], network[1])
+
+
+def check_route(router_name, network, expected_nexthop):
+    """
+    Checks if the given network is present on the given router and has the
+    expected next hop.
+    """
+    tgen = get_topogen()
+    router = tgen.gears[router_name]
+
+    expected_response = {
+        network: [
+            {
+                "prefix": network,
+                "protocol": "ospf",
+                "nexthops": [
+                    {
+                        "ip": expected_nexthop,
+                        "active": True,
+                    },
+                ],
+            },
+        ],
+    }
+
+    test_func = partial(
+        topotest.router_json_cmp,
+        router,
+        f"show ip route {network} json",
+        expected_response,
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=60, wait=1)
+    assert (
+        result is None
+    ), f"{router_name} (ospfd): no route {network} via {expected_nexthop}\n{dump_routes(router)}"
+
+    address = network.split("/")[0]
+    route = router.run(f"ip route get {address}")
+    assert (
+        f"via {expected_nexthop}" in route
+    ), f"{router_name} (kernel): no route {address} via {expected_nexthop}\n{dump_routes(router)}"
+
+
+def dump_routes(router):
```

Comment:
I think verify_rib() gives enough details when failing, I doubt this is useful to have.